src/components/search-filters/desktop/filters.tsx

```diff
@@ -1,105 +1,58 @@
+import { memo } from 'react';
 import { FiltersProps } from '../types';
 
 import { FilterGroup, FilterItem } from '@mdb/devcenter-components';
+import { CONTENT_TYPE_NAME_MAP } from '../../../data/constants';
+import { useCallback } from 'react';
 
-const DesktopFilters: React.FunctionComponent<FiltersProps> = ({
-    className,
-    onFilter,
-    allFilters,
-    l1Items,
-    languageItems,
-    technologyItems,
-    contributedByItems,
-    expertiseLevelItems,
-    contentTypeItems = [],
-    codeLevelItems = [],
-}) => {
-    const onToggle = (checked: boolean, filter: FilterItem) => {
-        if (checked) {
-            if (filter.subFilters?.length) {
-                const subFiltersToAdd = filter.subFilters.filter(
-                    subFilter =>
-                        !allFilters.find(
+const DesktopFilters: React.FunctionComponent<FiltersProps> = memo(
```

Comment:
I memo'd a few of these components related to this work because it actually does seem to cut down on unnecessary re-renders according to the React profiler 
<img width="537" alt="Screen Shot 2022-11-14 at 7 30 30 PM" src="https://user-images.githubusercontent.com/110849018/201796793-b86c3a6a-b06d-417c-8611-4b829af5d954.png">